### DISCUSSION TITLE
LuaSnip overrides buffer / tmux source?

### DISCUSSION THREAD

**Discussion by rodhash** [hrsh7th/nvim-cmp#2097] [General]
CMP is working great thanks but I noticed one thing: it seems LuaSnip is kind overriding completions from buffer and also tmux sources.

Is this expected?

Example:

I have `"desc"` in the line above but CMP only shows `"desc"` from LuaSnip not from Buffer:

<img width="787" alt="image" src="https://github.com/user-attachments/assets/83b1df58-bf5a-49fd-93ae-5d55c3074b7c">

After disabling LuaSnip source now I see `"desc"` completion coming from Buffer:

<img width="784" alt="image" src="https://github.com/user-attachments/assets/adee6ad2-cfd5-4403-901b-074cb6af28e5">

Bug or feature?

**Comment by Shougo** [comment]
https://github.com/hrsh7th/nvim-cmp/issues/511

Related?

**Reply by rodhash** [reply]
Yep that's it, thanks.. it seems CMP now handle duplicated entries automatically.
